auto_update.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+
+	"github.com/inconshreveable/go-update"
+)
+
+// RunUpdater check if an update is due, check if current version is outdated and perform update if needed
+func RunUpdater(app *TGFApplication) bool {
+	var autoUpdateFile = "tgfautoupdate"
+	var dueForUpdate = lastRefresh(autoUpdateFile) > 2*time.Hour
+	if !dueForUpdate {
+		app.Debug("Update not due")
+		return false
+	}
+	touchImageRefresh(autoUpdateFile)
+
+	v, err := getLatestVersion()
+	if err != nil {
+		app.Debug("Unable to fetch latest version from S3", err)
+		return false
+	}
+
+	latestVersion, err := semver.Make(v)
+	if err != nil {
+		app.Debug("Semver error", err)
```

Comment:
This should be a real error if that happens.

Reply:
Meaning, just a warning (more than debug)?
Or crashing the app?
Because auto updating should remain pretty silent I think

Reply:
Yes and no. Maybe not to the point of crashing the application. But the messages should be printed in red informing users that there is a problem. Not so much people set the -D flag and will see these messages if this situation happens.